src/main/webapp/app/forms/form-footer/form-footer.component.html

```diff
@@ -0,0 +1,32 @@
+<div class="update-programming-exercise-footer gap-4">
+    <jhi-exercise-update-notification
+        class="flex-grow-1"
+        [isCreation]="isCreation() ?? false"
+        [isImport]="isImport() ?? false"
+        [notificationText]="notificationText"
+        (notificationTextChange)="notificationTextChange.next($event)"
+    />
+    <div class="update-programming-exercise-footer-content">
+        @if (isSaving) {
+            <span jhiTranslate="artemisApp.quizExercise.edit.saving" class="badge bg-secondary"> </span>
+        }
+        <ng-template #tooltipTranslate>
+            @for (reason of invalidReasons; track reason) {
+                <div>
+                    <p jhiTranslate="{{ reason.translateKey }}"></p>
+                </div>
+            }
+        </ng-template>
+        @if (invalidReasons.length > 0) {
+            <span class="badge bg-danger" [ngbTooltip]="tooltipTranslate" tooltip-placement="top-right auto">
+                <fa-icon [icon]="faExclamationCircle" />
+                <span jhiTranslate="artemisApp.quizExercise.edit.invalidInput"></span>
+                <span>({{ invalidReasons.length }})</span>
+            </span>
+        }
+        <button type="button" id="cancel-save" class="btn btn-secondary me-1" (click)="cancel.next()">
+            <fa-icon [icon]="faBan" />&nbsp;<span jhiTranslate="entity.action.cancel">Cancel</span>
```

Comment:
Please remove the `&nbsp;` Just put the span into the next line. A newline will be automatically rendered as a space. 
```
<fa-icon [icon]="faBan" />
<span jhiTranslate="entity.action.cancel">Cancel</span>
```

See: https://docs.artemis.cit.tum.de/dev/guidelines/client.html#icons-with-text